editor/store/reducer.js

```diff
@@ -530,28 +524,29 @@ export function blockInsertionPoint( state = {}, action ) {
 export function preferences( state = PREFERENCES_DEFAULTS, action ) {
 	switch ( action.type ) {
 		case 'INSERT_BLOCKS':
-			// put the block in the recently used blocks
-			let recentlyUsedBlocks = [ ...state.recentlyUsedBlocks ];
-			action.blocks.forEach( ( block ) => {
-				recentlyUsedBlocks = [ block.name, ...without( recentlyUsedBlocks, block.name ) ].slice( 0, MAX_RECENT_BLOCKS );
-			} );
-			return {
-				...state,
-				recentlyUsedBlocks,
-			};
-		case 'SETUP_EDITOR':
-			const isBlockDefined = name => getBlockType( name ) !== undefined;
-			const filterInvalidBlocksFromList = list => list.filter( isBlockDefined );
-			const commonBlocks = getBlockTypes()
-				.filter( ( blockType ) => 'common' === blockType.category )
-				.map( ( blockType ) => blockType.name );
+			return action.blocks.reduce( ( prevState, block ) => {
+				let insert;
+				if ( isReusableBlock( block ) ) {
+					insert = { name: block.name, ref: block.attributes.ref };
+				} else {
+					insert = { name: block.name };
+				}
```

Comment:
minor: maybe just my obsession for shorter code. This could be:

```js
const insert = { name : block.name };
if ( isReusableBlock( block ) ) {
  insert.ref = block.attributes.ref;
}
```

Reply:
Changed in 6d77903e3a82128cc77adf86c997ee3131f0e6f0. All this newfangled syntax has made me forget that mutation is still a thing 😆